### ISSUE TITLE
Gatsby graphql.config.json is not found

### ISSUE THREAD

**Issue by vepanimas** [JetBrains/js-graphql-intellij-plugin#683]
Schema is not discovered for Gatsby projects when `graphqlTypegen: true` is enabled in `gatsby-config.js`.

Reported here https://github.com/JetBrains/js-graphql-intellij-plugin/issues/673#issuecomment-1674947544.

**Comment by arm1n** [comment]
Hey @vepanimas thanks for providing a fix for this, I can confirm that it's working with the `4.0.2` and adding a `graphql.config.js` in the root of the project.

**Comment by robclancy** [comment]
This is not fixed in the released 4.0.2.

**Comment by vepanimas** [comment]
> It's not working when the folder is excluded... it shouldn't be a surprise that an auto generated file that only works with a specific plugin might be excluded.

@robclancy Why do you think it should be excluded?

**Comment by robclancy** [comment]
It's a cache and tmp folder...

**Comment by vepanimas** [comment]
@robclancy Yep, it's correct, though, why do you need to exclude it? You usually exclude a folder if you don't want the files inside of it to be indexed and to appear in completion, resolve, navigation, etc. Nevertheless, we do have an [issue](https://github.com/JetBrains/js-graphql-intellij-plugin/issues/334) related to that, but it's not yet clear when it will be implemented.

**Comment by robclancy** [comment]
What are you talking about? A tmp and cache folder is always excluded. A tmp folder is used for generated files. This issue is not resolved because if you exclude the folder that you should be excluding, the entire reason for the exclude feature to exist, then gatsby graphql config does not work.

**Comment by vepanimas** [comment]
> What are you talking about?

Look, when you want a file to be excluded from indexing, you mark it or the whole folder as excluded. That's what is written in the docs I provided. For some technologies, e.g. node.js and its node_modules, we have some workarounds, that allow us to use excluded files for completion and resolve, but usually, when you exclude something, it's completely ignored by the IDE.

The easiest way for now to bypass that and leave the `.cache` folder excluded is to un-exclude a specific directory you need, for Gatsby it is `.cache/typegen`.

https://github.com/JetBrains/js-graphql-intellij-plugin/assets/10243762/38396882-baac-4029-b937-f73c03747258

**Comment by robclancy** [comment]
Nothing you have said has contradicted anything I have said. I have no idea why you are saying these things.
